include/mbedtls/pk.h

```diff
@@ -280,6 +280,45 @@ void mbedtls_pk_restart_free(mbedtls_pk_restart_ctx *ctx);
  */
 int mbedtls_pk_setup(mbedtls_pk_context *ctx, const mbedtls_pk_info_t *info);
 
+/**
+ * \brief Initialize a PK context to wrap a PSA key.
+ *
+ * This function creates a PK context which wraps a PSA key. The PSA wrapped
+ * key must be an EC or RSA key pair (DH is not suported in the PK module).
+ *
+ * Under the hood PSA functions will be used to perform the required
+ * operations and, based on the key type, used algorithms will be:
+ * * EC:
+ *     * verify, verify_ext, sign, sign_ext: ECDSA.
+ * * RSA:
+ *     * sign: use the primary algorithm in the wrapped PSA key;
+ *     * sign_ext: RSA PSS if the pk_type is #MBEDTLS_PK_RSASSA_PSS, otherwise
+ *       it falls back to the sign() case;
+ *     * verify, verify_ext: not supported.
+ *
+ * In order for the above operations to succeed, the policy of the wrapped PSA
+ * key must allow the specified algorithm.
+ *
+ * Opaque PK contexts wrapping an EC keys also support \c mbedtls_pk_check_pair(),
```

Comment:
Along with renaming the function, we need to update the documentation to talk of “wrapped keys” rather than “opaque keys”.

```suggestion
 * PK contexts wrapping an EC keys also support \c mbedtls_pk_check_pair(),
```

Also please update other occurrences of “opaque” in this file (we'll deal with other documentation such as the transition guide as part of other tasks):

* “opaque signature mechanism (PSA)” → “PSA signature mechanism”
* “Opaque keys” → “Wrapped keys”
* Old function name referenced elsewhere.

Reply:
I've updated the references in pk.h, I've left the references in mbedtls_pk_setup_opaque as I believe these are currently out of scope of this task. Let me know if this is not the case though.